cupy_backends/cuda/libs/cusparselt.pyx

```diff
@@ -0,0 +1,365 @@
+        void* workspace, driver.Stream* streams, int32_t numStreams)
+
+    # Helper Functions
+    cusparseStatus_t cusparseLtSpMMAPrune(
+        const cusparseLtHandle_t* handle,
+        const cusparseLtMatmulDescriptor_t* matmulDescr,
+        const void* d_in, void* d_out,
+        cusparseLtPruneAlg_t pruneAlg, driver.Stream stream)
+    cusparseStatus_t cusparseLtSpMMAPruneCheck(
+        const cusparseLtHandle_t* handle,
+        const cusparseLtMatmulDescriptor_t* matmulDescr,
+        const void* d_in, int* valid, driver.Stream stream)
+    cusparseStatus_t cusparseLtSpMMACompressedSize(
+        const cusparseLtHandle_t* handle, const cusparseLtMatmulPlan_t* plan,
+        size_t* compressedSize)
+    cusparseStatus_t cusparseLtSpMMACompress(
+        const cusparseLtHandle_t* handle, const cusparseLtMatmulPlan_t* plan,
+        const void* d_dense, void* d_compressed, driver.Stream stream)
+
+
+###############################################################################
+# Classes
+###############################################################################
+
+cdef class Handle:
+    cdef void * _ptr
+
+    def __init__(self):
+        self._ptr = PyMem_Malloc(sizeof(cusparseLtHandle_t))
+
+    def __dealloc__(self):
+        PyMem_Free(self._ptr)
+        self._ptr = NULL
+
+    @property
+    def ptr(self):
+        return <intptr_t>self._ptr
+
+
+cdef class MatDescriptor:
+    cdef void * _ptr
+
+    def __init__(self):
+        self._ptr = PyMem_Malloc(sizeof(cusparseLtMatDescriptor_t))
+
+    def __dealloc__(self):
+        PyMem_Free(self._ptr)
+        self._ptr = NULL
+
+    @property
+    def ptr(self):
+        return <intptr_t>self._ptr
+
+
+cdef class MatmulDescriptor:
+    cdef void * _ptr
+
+    def __init__(self):
+        self._ptr = PyMem_Malloc(sizeof(cusparseLtMatmulDescriptor_t))
+
+    def __dealloc__(self):
+        PyMem_Free(self._ptr)
+        self._ptr = NULL
+
+    @property
+    def ptr(self):
+        return <intptr_t>self._ptr
+
+
+cdef class MatmulAlgSelection:
+    cdef void * _ptr
+
+    def __init__(self):
+        self._ptr = PyMem_Malloc(sizeof(cusparseLtMatmulAlgSelection_t))
+
+    def __dealloc__(self):
+        PyMem_Free(self._ptr)
+        self._ptr = NULL
+
+    @property
+    def ptr(self):
+        return <intptr_t>self._ptr
+
+
+cdef class MatmulPlan:
+    cdef void * _ptr
+
+    def __init__(self):
+        self._ptr = PyMem_Malloc(sizeof(cusparseLtMatmulPlan_t))
+
+    def __dealloc__(self):
+        PyMem_Free(self._ptr)
+        self._ptr = NULL
+
+    @property
+    def ptr(self):
+        return <intptr_t>self._ptr
+
+
+###############################################################################
+# Error handling
+###############################################################################
+
+@cython.profile(False)
+cpdef inline check_status(int status):
+    if status != 0:
+        raise cusparse.CuSparseError(status)
```

Comment:
Just checking: the error codes used in cuSPARSELt are the same as those in cuSPARSE?

Reply:
Yes, that's right, the return type of the cuSPARSELt functions is `cusparseStatus_t`.
https://docs.nvidia.com/cuda/cusparselt/functions.html